nuklear_console_textedit.h

```diff
@@ -0,0 +1,366 @@
+NK_API void nk_console_textedit_button_main_click(nk_console* button);
+NK_API void nk_console_textedit_button_back_click(nk_console* button);
+
+#if defined(__cplusplus)
+}
+#endif
+
+#endif // NK_CONSOLE_TEXTEDIT_H__
+
+#if defined(NK_CONSOLE_IMPLEMENTATION) && !defined(NK_CONSOLE_HEADER_ONLY)
+#ifndef NK_CONSOLE_TEXTEDIT_IMPLEMENTATION_ONCE
+#define NK_CONSOLE_TEXTEDIT_IMPLEMENTATION_ONCE
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+/**
+ * Handle the click event for textedit's children items.
+ */
+NK_API void nk_console_textedit_button_back_click(nk_console* button) {
+    // Make sure we're not going back to a Row.
+    if (button->parent->type == NK_CONSOLE_ROW) {
+        button = button->parent;
+    }
+
+    nk_console_button_back(button);
+
+    // Clear out the on-screen keyboard
+    nk_console* enter_textedit_button = button->parent;
+    nk_console_free_children(enter_textedit_button);
+}
+
+#include <stdio.h>
+NK_API void nk_console_textedit_key_click(nk_console* key) {
+    // Find the textedit widget, and make sure it's not a row.
+    nk_console* textedit = key->parent;
+    if (textedit->type == NK_CONSOLE_ROW) {
+        textedit = textedit->parent;
+    }
+
+    // Get the textedit data.
+    nk_console_textedit_data* data = (nk_console_textedit_data*)textedit->data;
+
+    // Handle the key press
+    enum nk_symbol_type symbol = nk_console_button_get_symbol(key);
+    switch (symbol) {
+        case NK_SYMBOL_CIRCLE_SOLID:
+        case NK_SYMBOL_CIRCLE_OUTLINE:
+            data->shift = !data->shift;
+            nk_console_button_set_symbol(key, symbol == NK_SYMBOL_CIRCLE_SOLID ? NK_SYMBOL_CIRCLE_OUTLINE : NK_SYMBOL_CIRCLE_SOLID);
+
+
+
+            // Replace all labels of the parent buttons with shifted characters.
+            for (size_t x = 0; x < cvector_size(textedit->children); ++x) {
+                nk_console* child = textedit->children[x];
+                if (child->type == NK_CONSOLE_ROW) {
+                    nk_console_row* row = (nk_console_row*)child;
+                    for (size_t i = 0; i < cvector_size(row->data.children); ++i) {
+                        nk_console* activeButton = row->data.children[i].console;
+                        if (activeButton->type == NK_CONSOLE_BUTTON) {
+                            const char* label = activeButton->label;
+                            if (label != NULL && nk_strlen(label) == 1) {
+                                char old_char = label[0];
+                                switch (old_char) {
+                                    case 'a': label = "A"; break;
+                                    case 'b': label = "B"; break;
+                                    case 'c': label = "C"; break;
+                                    case 'd': label = "D"; break;
+                                    case 'e': label = "E"; break;
+                                    case 'f': label = "F"; break;
```

Comment:
This is definitely not the best solution, but I'm unsure how to handle it, since the button does not own its label data.

Reply:
Longer term I think there should be a way to customize the buttons on the keyboard (esp for non-english languages) but this is good enough for the first pass!